### PR TITLE
Ongoing fixes

### LINKED ISSUE #352: DateTime type with timezone is interpreted as String by Tableau
ClickHouse server version 21.2.3 revision 54447
ClickHouse ODBC version v1.1.9.20201226
Tableau Desktop version 2020.4

```sql
CREATE DATABASE test;
create table test.order ( order_datetime DateTime('UTC'), order_datetime_without_timezon DateTime ) ENGINE = MergeTree() ORDER BY (order_datetime);
INSERT INTO test.order VALUES ('2021-01-01 00:03:00', '2021-01-01 00:03:00');
```
DateTime without timezone is interpreted correctly.

image

Thank you.

### LINKED ISSUE #353: Error while using Null values in Tableau
Hi! We found some error, while using Null values in Tableau.

We get this error in release https://github.com/ClickHouse/clickhouse-odbc/releases/tag/v1.1.9.20201226, but
in release https://github.com/ClickHouse/clickhouse-odbc/releases/tag/v1.1.1.20191108 everything works just fine.

An error occurred while communicating with the Other Databases (ODBC) data source 'oroboros_test_123 (cubes)'
Bad Connection: Tableau could not connect to the data source.
Error Code: B19090E0
unknown type
SELECT SUM({fn CONVERT(NULL, SQL_BIGINT)}) AS `sum_Calculation_3063644052176396288_ok`
FROM `oroboros_test_123`
HAVING (COUNT(1) > 0)

This error occurs while creating Calculation Filed like `IF FALSE THEN 1 END`
How can we fix this?

UPD:
I changed some code here https://github.com/ClickHouse/clickhouse-odbc/blob/3c37dbd1e4e75cfba8dfa5300c52fcfd0d94a008/driver/utils/type_info.h#L51, by replacing this line to throw std::runtime_error("unknown type. Got type " + type);
And got next message

An error occurred while communicating with the Other Databases (ODBC) data source 'cubes_clickhouse__fact_ue_partitioned_distributed (cubes)'
Bad Connection: Tableau could not connect to the data source.
Error Code: B19090E0
unknown type. Got type Nothing
SELECT SUM({fn CONVERT(NULL, SQL_BIGINT)}) AS sum_Calculation_3485082461064859648_ok
FROM cubes_clickhouse__fact_ue_partitioned_distributed
HAVING (COUNT(1) > 0)

As far as I can see, there is no type `Nothing` here https://github.com/ClickHouse/clickhouse-odbc/blob/3c37dbd1e4e75cfba8dfa5300c52fcfd0d94a008/driver/utils/type_info.cpp#L7, 
but there is type Nothing in Clickhouse doc https://clickhouse.tech/docs/ru/sql-reference/data-types/special-data-types/nothing/

### LINKED ISSUE #354: Driver doesnt generate error if you try to create connection with wrong credentials
Hi!

I'm trying to connect from Tableau Desktop using ODBC driver. 
If I enter wrong credentials - driver doesn't recognize this situation and doesn't generate error/exception.
As result Tableau assumes that connection was successful and opens UI with empty tables list.
This is confusing for end users.

Driver: 1.1.9.20201226
Tableau Desktop : 2020.4

### LINKED ISSUE #355: error when try cmake --build . --config RelWithDebInfo on MAC OS BIG SUR 11.3.1
[ 16%] Built target Foundation
[ 65%] Built target crypto
[ 68%] Built target ssl
[ 70%] Built target Crypto
[ 76%] Built target XML
[ 78%] Built target JSON
[ 81%] Built target Util
[ 90%] Built target Net
[ 91%] Built target NetSSL
[ 93%] Built target clickhouse-odbc-impl
[ 95%] Built target clickhouse-odbcw-impl
[ 95%] Built target clickhouse-odbcw
[ 95%] Built target clickhouse-odbc
[ 95%] Built target clickhouse-odbc-load-ut
Consolidate compiler generated dependencies of target nanodbc
[ 95%] Building CXX object contrib/nanodbc/CMakeFiles/nanodbc.dir/nanodbc/nanodbc.cpp.o
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:79:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/cstddef:37:
/Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/version:1:1: error: expected unqualified-id
2.13.0
^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:79:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/cstddef:49:9: error: no member named 'ptrdiff_t' in the global namespace
using ::ptrdiff_t;
      ~~^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:501:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/type_traits:418:
/Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/version:1:1: error: expected unqualified-id
2.13.0
^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:502:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/typeinfo:60:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/exception:83:
/Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/version:1:1: error: expected unqualified-id
2.13.0
^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:504:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:671:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/new:93:
/Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/version:1:1: error: expected unqualified-id
2.13.0
^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:504:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:672:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/utility:205:
/Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/version:1:1: error: expected unqualified-id
2.13.0
^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:504:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:673:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/limits:121:
/Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/version:1:1: error: expected unqualified-id
2.13.0
^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:504:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:674:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/iterator:423:
/Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/version:1:1: error: expected unqualified-id
2.13.0
^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:504:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:674:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/iterator:562:13: error: unknown type name 'ptrdiff_t'
    typedef ptrdiff_t difference_type;
            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/iterator:623:56: error: unknown type name 'ptrdiff_t'
template<class _Category, class _Tp, class _Distance = ptrdiff_t,
                                                       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/iterator:989:66: error: unknown type name 'ptrdiff_t'
          class _Traits = char_traits<_CharT>, class _Distance = ptrdiff_t>
                                                                 ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/iterator:989:54: error: template parameter missing a default argument
          class _Traits = char_traits<_CharT>, class _Distance = ptrdiff_t>
                                                     ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/iterator:989:27: note: previous default template argument defined here
          class _Traits = char_traits<_CharT>, class _Distance = ptrdiff_t>
                          ^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:504:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:677:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/tuple:154:
/Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/version:1:1: error: expected unqualified-id
2.13.0
^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:504:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:681:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/atomic:576:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/__threading_support:14:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/chrono:831:
/Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/version:1:1: error: expected unqualified-id
2.13.0
^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:504:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:681:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/atomic:581:
/Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/version:1:1: error: expected unqualified-id
2.13.0
^
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.cpp:16:
In file included from /Users/yurchyshynd/Desktop/odbc/clickhouse-odbc/contrib/nanodbc/nanodbc/nanodbc.h:80:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/functional:504:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:681:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/atomic:1052:73: error: unknown type name 'ptrdiff_t'
_Tp* __cxx_atomic_fetch_add(__cxx_atomic_base_impl<_Tp*> volatile* __a, ptrdiff_t __delta, memory_order __order) _NOEXCEPT {
                                                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/atomic:1057:65: error: unknown type name 'ptrdiff_t'
_Tp* __cxx_atomic_fetch_add(__cxx_atomic_base_impl<_Tp*> * __a, ptrdiff_t __delta, memory_order __order) _NOEXCEPT {
                                                                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/atomic:1073:73: error: unknown type name 'ptrdiff_t'
_Tp* __cxx_atomic_fetch_sub(__cxx_atomic_base_impl<_Tp*> volatile* __a, ptrdiff_t __delta, memory_order __order) _NOEXCEPT {
                                                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/atomic:1078:65: error: unknown type name 'ptrdiff_t'
_Tp* __cxx_atomic_fetch_sub(__cxx_atomic_base_impl<_Tp*> * __a, ptrdiff_t __delta, memory_order __order) _NOEXCEPT {
                                                                ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
gmake[2]: *** [contrib/nanodbc/CMakeFiles/nanodbc.dir/build.make:76: contrib/nanodbc/CMakeFiles/nanodbc.dir/nanodbc/nanodbc.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2119: contrib/nanodbc/CMakeFiles/nanodbc.dir/all] Error 2
gmake: *** [Makefile:166: all] Error 2

### LINKED ISSUE #361: Unsupported type NULL
Hello.
I've set up odbc driver on Linux (Ubuntu 18.04). 
I am testing it via isql terminal, and I can't execute simple query like "SELECT NULL as test;" (see screenshot).
Is it possible to work with null objects or should I use other types?
image

### LINKED ISSUE #364: data type datetime64 shows up as a string in Tableau (using Clikchouse ODBC driver and connector)
image

### LINKED ISSUE #365: UInt64 is not recognized to Tableau
Tableau generates error if you try to use open table with UInt64 type.

Error is: An error occurred while loading the data. Cannot interpret '18150585289071012696' as signed 64-bit integer: stoll argument out of range

Steps to reproduce:
Connect to ClickHouse Playground database from Tableau and try to open "hits_100m_obfuscated" table.

### PR DESCRIPTION
**Pull request by traceon**
- Fix/proofread/extend the `README.md`
- Closes #355
- Closes #361
- Closes #353
- Closes #364
- Closes #352
- Closes #354 - added `VerifyConnectionEarly` functionality to enable extra check on `SQLConnect()`
- Closes #365 - added `HugeIntAsString` functionality

### CODE REVIEW

**Review by Enmk — APPROVED** [review]
There are some minor questions, but overall looks good.

File: driver/connection.cpp [hunk 1]
```diff
@@ -53,6 +53,20 @@ Connection::Connection(Environment & environment)
     resetConfiguration();
 }
 
+const TypeInfo & Connection::getTypeInfo(const std::string & type_name, const std::string & type_name_without_parameters) const {
+    auto tmp_type_name = type_name;
+    auto tmp_type_name_without_parameters = type_name_without_parameters;
+
+    const auto tmp_type_without_parameters_id = convertUnparametrizedTypeNameToTypeId(tmp_type_name_without_parameters);
+
+    if (huge_int_as_string && tmp_type_without_parameters_id == DataSourceTypeId::UInt64) {
```
Comment (Enmk) [inline driver/connection.cpp:62]: What about other wide types, like `Int128`, `UInt128`, `Decimal128` and wider?

File: driver/format/RowBinaryWithNamesAndTypes.cpp [hunk 2]
```diff
@@ -152,14 +153,24 @@ void RowBinaryWithNamesAndTypesResultSet::readValue(WireTypeDateTimeAsInt & dest
     readPOD(dest.value);
 }
 
+void RowBinaryWithNamesAndTypesResultSet::readValue(WireTypeDateTime64AsInt & dest, ColumnInfo & column_info) {
+    readPOD(dest.value);
+}
+
 void RowBinaryWithNamesAndTypesResultSet::readValue(DataSourceType<DataSourceTypeId::Date> & dest, ColumnInfo & column_info) {
-    WireTypeDateAsInt dest_raw;
+    WireTypeDateAsInt dest_raw(column_info.timezone);
```
Comment (Enmk) [inline driver/format/RowBinaryWithNamesAndTypes.cpp:161]: `Date` has no timezone in CH

File: driver/format/RowBinaryWithNamesAndTypes.cpp [hunk 3]
```diff
@@ -152,14 +153,24 @@ void RowBinaryWithNamesAndTypesResultSet::readValue(WireTypeDateTimeAsInt & dest
     readPOD(dest.value);
 }
 
+void RowBinaryWithNamesAndTypesResultSet::readValue(WireTypeDateTime64AsInt & dest, ColumnInfo & column_info) {
+    readPOD(dest.value);
```
Comment (Enmk) [inline driver/format/RowBinaryWithNamesAndTypes.cpp:157]: Did you forget to use\convert value according to a timezone?

File: driver/test/datetime_it.cpp [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+#include "driver/test/client_test_base.h"
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <iostream>
+
+class DateTime
+    : public ClientTestWithParamBase<
+        std::tuple<
```
Comment (Enmk) [inline driver/test/datetime_it.cpp:12]: IMO it is better to use named struct here, it is easier to read and reason about.

Comment (Enmk) [inline driver/test/datetime_it.cpp:12]: IMO it is better to use named struct here, it is easier to read and reason about.

File: driver/connection.cpp [hunk 5]
```diff
@@ -445,6 +456,13 @@ void Connection::setConfiguration(const key_value_map_t & cs_fields, const key_v
         stringmaxlength = TypeInfo::string_max_size;
 }
 
+void Connection::verifyConnection() {
+    LOG("Verifying connection and credentials...");
+    auto & statement = allocateChild<Statement>();
+    statement.executeQuery("SELECT 1");
```
[the remark this answers is not in the document]
Comment (traceon) [reply driver/connection.cpp:483]: Yeah, I should have added a proper cleanup there. Will fix.

File: driver/format/RowBinaryWithNamesAndTypes.cpp (hunk 2 above)
Comment (traceon) [reply driver/format/RowBinaryWithNamesAndTypes.cpp:161]: Yeah, but the timezone (of the data/server) might be used when it is converted (read by the user) to some DateTime structures.

File: driver/format/RowBinaryWithNamesAndTypes.cpp (hunk 3 above)
Comment (traceon) [reply driver/format/RowBinaryWithNamesAndTypes.cpp:157]: It should be done later on the call stack (`TODO` yes, not the goal of this PR):

https://github.com/traceon/clickhouse-odbc/blob/67b36ef86ebea0fd5372b219f5c84c9771b1eacc/driver/utils/type_info.h#L1871-L1901

File: driver/test/datetime_it.cpp (hunk 4 above)
Comment (traceon) [reply driver/test/datetime_it.cpp]: Changed.

**Comment by traceon** [comment]
GCC builds in macOS are failing due to nanodbc - a fix for that was proposed in their repo, once (and if) it is merged and submodule here is bumped, these failures will be gone. For now, it is fairly safe to merge with those builds failing.

**Pull request merged**